helpers/build_packages.sh

```diff
@@ -151,7 +151,60 @@ if [ "$BUILDMW" = "1" ]; then
 
     pushd $ANDROID_ROOT/hybris/mw > /dev/null
 
-    if [ "$BUILDMW_REPO" = "" ]; then
+    MANIFEST_LOOKUP=$ANDROID_ROOT
+    while true; do
+        MANIFEST=$MANIFEST_LOOKUP/.repo/manifest.xml
+        if [ -f "$MANIFEST" ] || [ "$MANIFEST_LOOKUP" == "$(dirname "$MANIFEST_LOOKUP")" ]; then
```

Comment:
If you don't use bash features you can still use [ var = var ] here.
Also dirname can be replaced with ${var%/*}, the intentions isn't that obvious this way but it avoid external commands and command substitution.

Reply:
Looks ok but reduces readability. There are lots of other shell substitutions, so let's leave it at that.